app/controllers/calendars_controller.rb

```
@@ -46,8 +46,20 @@ def create
   end
 
   def edit
-    response = @client.edit_calendar(params[:id])
-    render locals:{calendar: CalendarPresenter.new(response)}
+    calendar = Calendar.find_by(google_id: decode_id(params[:id]))
+    if current_user.id == calendar.owner_id
+      render locals:{calendar: calendar}
+    else
+      redirect_to calendars_path
+    end
+  end
+
+  def update
+    calendar = Calendar.find_by(google_id: decode_id(params[:id]))
+    calendar.update(summary: params[:summary])
```

try using the `!` variant everywhere. and thing about the failure cases
if we use the `update!` atleast the code will not silently fail